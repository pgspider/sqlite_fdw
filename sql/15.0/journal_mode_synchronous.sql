CREATE EXTENSION sqlite_fdw;

--Testcase 1: read the content inserted during the initialisation of tests
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw
OPTIONS (database '/tmp/sqlitefdw_test_core.db');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
select * from test_pragma_jms;
DROP SERVER server CASCADE;

--Testcase 2: tests for journal_mode 'delete'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw
OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'delete');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
INSERT INTO test_pragma_jms (attr) VALUES (1);

DROP SERVER server CASCADE;

--Testcase 3: tests for journal_mode 'truncate'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'truncate');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
INSERT INTO test_pragma_jms (attr) VALUES (2);

DROP SERVER server CASCADE;

--Testcase 4: tests for journal_mode 'persist'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'persist');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
INSERT INTO test_pragma_jms (attr) VALUES (3);

DROP SERVER server CASCADE;

--Testcase 5: tests for journal_mode 'memory'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'memory');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
INSERT INTO test_pragma_jms (attr) VALUES (4);

DROP SERVER server CASCADE;

--Testcase 6: tests for journal_mode 'wal'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'wal');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
INSERT INTO test_pragma_jms (attr) VALUES (5);

DROP SERVER server CASCADE;

--Testcase 7: tests for journal_mode 'off'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'off');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
INSERT INTO test_pragma_jms (attr) VALUES (6);

DROP SERVER server CASCADE;

--Testcase 8: tests for wrong value of journal_mode
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'WrongValue');

--Testcase 9: tests for synchronous 'off'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'off');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (7);
SELECT * FROM test_pragma_jms;

DROP SERVER server CASCADE;

--Testcase 10: tests for synchronous 'normal'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'normal');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (8);
SELECT * FROM test_pragma_jms;

DROP SERVER server CASCADE;

--Testcase 11: tests for synchronous 'full'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'full');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (9);
SELECT * FROM test_pragma_jms;

DROP SERVER server CASCADE;

--Testcase 12: tests for synchronous 'extra'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'extra');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (10);
SELECT * FROM test_pragma_jms;

DROP SERVER server CASCADE;

--Testcase 13: tests for both journal_mode and synchronous
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'wal', synchronous 'normal');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (11);
SELECT * FROM test_pragma_jms;

DROP SERVER server CASCADE;

--Testcase 14: tests for wrong value of synchronous
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'WrongValue');
DROP EXTENSION sqlite_fdw cascade;
