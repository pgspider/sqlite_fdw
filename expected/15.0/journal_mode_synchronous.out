CREATE EXTENSION sqlite_fdw;
--Testcase 1: read the content inserted during the initialisation of tests
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw
OPTIONS (database '/tmp/sqlitefdw_test_core.db');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
select * from test_pragma_jms;
 attr 
------
    0
(1 row)

DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 2: tests for journal_mode 'delete'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw
OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'delete');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
 attr 
------
    0
(1 row)

INSERT INTO test_pragma_jms (attr) VALUES (1);
DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 3: tests for journal_mode 'truncate'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'truncate');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
(2 rows)

INSERT INTO test_pragma_jms (attr) VALUES (2);
DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 4: tests for journal_mode 'persist'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'persist');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
(3 rows)

INSERT INTO test_pragma_jms (attr) VALUES (3);
DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 5: tests for journal_mode 'memory'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'memory');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
    3
(4 rows)

INSERT INTO test_pragma_jms (attr) VALUES (4);
DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 6: tests for journal_mode 'wal'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'wal');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
    3
    4
(5 rows)

INSERT INTO test_pragma_jms (attr) VALUES (5);
DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 7: tests for journal_mode 'off'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'off');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
    3
    4
    5
(6 rows)

INSERT INTO test_pragma_jms (attr) VALUES (6);
DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 8: tests for wrong value of journal_mode
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'WrongValue');
ERROR:  "journal_mode" parameter is not valid; accepted values: 'delete', 'truncate', 'persist', 'memory', 'wal', 'off'.
--Testcase 9: tests for synchronous 'off'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'off');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (7);
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
    3
    4
    5
    6
    7
(8 rows)

DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 10: tests for synchronous 'normal'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'normal');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (8);
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
    3
    4
    5
    6
    7
    8
(9 rows)

DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 11: tests for synchronous 'full'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'full');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (9);
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
    3
    4
    5
    6
    7
    8
    9
(10 rows)

DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 12: tests for synchronous 'extra'
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'extra');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (10);
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
    3
    4
    5
    6
    7
    8
    9
   10
(11 rows)

DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 13: tests for both journal_mode and synchronous
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', journal_mode 'wal', synchronous 'normal');
CREATE FOREIGN TABLE test_pragma_jms (
  attr int
) SERVER server;
INSERT INTO test_pragma_jms (attr) VALUES (11);
SELECT * FROM test_pragma_jms;
 attr 
------
    0
    1
    2
    3
    4
    5
    6
    7
    8
    9
   10
   11
(12 rows)

DROP SERVER server CASCADE;
NOTICE:  drop cascades to foreign table test_pragma_jms
--Testcase 14: tests for wrong value of synchronous
CREATE SERVER server FOREIGN DATA WRAPPER sqlite_fdw 
   	OPTIONS (database '/tmp/sqlitefdw_test_core.db', synchronous 'WrongValue');
ERROR:  "synchronous" parameter is not valid; accepted values: 'off', 'normal', 'full', 'extra'.
DROP EXTENSION sqlite_fdw cascade;
